helix-view/src/editor.rs

```diff
@@ -571,7 +577,30 @@ impl Editor {
         self.status_msg = Some((error.into(), Severity::Error));
     }
 
+    pub fn unset_theme_preview(&mut self) {
+        if let Some(last_theme) = self.last_theme.take() {
+            let theme_name = &(*last_theme).name;
+            match self.theme_loader.load(theme_name) {
+                Ok(theme) => {
+                    self.set_theme(theme);
+                }
+                Err(_e) => {
+                    self.set_error(format!("Failed to set theme: '{}'", theme_name));
+                }
+            };
+        }
+    }
+
+    pub fn set_theme_preview(&mut self, theme: Theme) {
+        self.set_theme_impl(theme);
+    }
+
     pub fn set_theme(&mut self, theme: Theme) {
+        self.last_theme = Some(Box::new(theme.clone()));
+        self.set_theme_impl(theme);
+    }
```

Comment:
```suggestion
    pub fn set_theme_preview(&mut self, theme: Theme) {
        self.last_theme = Some(self.theme.replace(theme));
        self.set_theme_impl(self.theme);
    }

    pub fn set_theme(&mut self, theme: Theme) {
        self.last_theme = None;
        self.set_theme_impl(theme);
    }
```

Reply:
With regards to:

```rust
self.last_theme = Some(self.theme.replace(theme));
```
do you mean `std::mem::replace`?

```rust
self.last_theme = Some(std::mem::replace(&mut self.theme, theme));
```

Also, the reason I've not set `last_theme` in `set_theme_preview` is because `last_theme` doesn't represent "the last theme that was shown in the preview", but rather, "the last theme that was selected in the editor". This means I only want to set `last_theme` once during preview at the beginning: it sets the `last_theme` to the already existing theme so that if a user cancels while previewing, it goes back to their originally selected theme as opposed to the last previewed theme.

Reply:
I sort of worked it out in a similar way! I moved it into the `impl` so I could `replace` without running into _borrowing mutable -> borrowing immutable_, and I added a guard for handling the first theme preview